### PR TITLE
Add some documentation describing how to interpret the symbols shown during recompilation

### PR DESCRIPTION
**Pull request by pbouffard**
I started noticing the `?` indicator show up in Julia 1.10 which I assume relates to parallel precompilation. I'm not exactly sure what it means nor also by what logic packages finished with precompilation (marked `✓`) disappear from the list (I'm guessing that all their transitive dependencies are finished?).

This seemed to me to be the best place in the docs to put this.

### CODE REVIEW

File: docs/src/environments.md [hunk 1]
```diff
@@ -192,6 +192,14 @@ force these packages to be retried, as `pkg> precompile` will always retry all p
 
 To disable the auto-precompilation, set `ENV["JULIA_PKG_PRECOMPILE_AUTO"]=0`.
 
+The indicators next to the package names displayed during precompilation
+indicate the status of that package's precompilation. A `?` character indicates
+that [TODO: something to do with waiting?]; animated "clock" characters
```
Comment (IanButterworth) [inline docs/src/environments.md:197]:
```suggestion
that a `PrecompilableError` was thrown which indicates precompilation was disallowed i.e. `__precompile__(false)`; animated "clock" characters
```

**Comment by IanButterworth** [comment]
Maybe these are better as a bullet list?

Also can you add detail about yellow check marks, which indicate packages that have been precompiled, but different versions are currently loaded, so cannot be used until julia is restarted.

**Comment by IanButterworth** [comment]
bump

File: docs/src/environments.md [hunk 2]
```diff
@@ -192,6 +192,14 @@ force these packages to be retried, as `pkg> precompile` will always retry all p
 
 To disable the auto-precompilation, set `ENV["JULIA_PKG_PRECOMPILE_AUTO"]=0`.
 
+The indicators next to the package names displayed during precompilation
+indicate the status of that package's precompilation. Animated "clock" characters
+(`◐`,`◓`,`◑`,`◒`) indicate that the package is currently being precompiled.
+A checkmark (`✓`) indicates that the package has been successfully precompiled (after which that package will disappear from the list).
+A question mark (`?`) character indicates that a `PrecompilableError` was thrown, indicating that precompilation was disallowed,
+i.e. `__precompile__(false)` in that package.
+An `✗` indicates that the package failed to precompile.
```
Comment (asinghvi17) [inline docs/src/environments.md:201]:
```suggestion
The indicators next to the package names displayed during precompilation
indicate the status of that package's precompilation. 

- `[◐, ◓, ◑, ◒]` Animated "clock" characters indicate that the package is currently being precompiled.
- `✓` A checkmark indicates that the package has been successfully precompiled (after which that package will disappear from the list).
- `?` A question mark character indicates that a `PrecompilableError` was thrown, indicating that precompilation was disallowed, i.e. `__precompile__(false)` in that package.
- `✗` A cross indicates that the package failed to precompile.

File: docs/src/environments.md [hunk 3]
```diff
@@ -192,6 +192,14 @@ force these packages to be retried, as `pkg> precompile` will always retry all p
 
 To disable the auto-precompilation, set `ENV["JULIA_PKG_PRECOMPILE_AUTO"]=0`.
 
+The indicators next to the package names displayed during precompilation
+indicate the status of that package's precompilation. 
+
+- `[◐, ◓, ◑, ◒]` Animated "clock" characters indicate that the package is currently being precompiled.
+- `✓` A checkmark indicates that the package has been successfully precompiled (after which that package will disappear from the list).
```
Comment (IanButterworth) [inline docs/src/environments.md:199]:
```suggestion
- `✓` A green checkmark indicates that the package has been successfully precompiled (after which that package will disappear from the list). If the checkmark is yellow it means that the package is currently loaded so the session will need to be restarted to access the version that was just precompiled.
```

**Pull request merged**